docs/source/reference-core.rst

```diff
@@ -768,43 +767,91 @@ inside the handler function(s) with the ``nonlocal`` keyword::
-
-Ideally, in some abstract sense we'd want everything that *can* raise an
-`ExceptionGroup` to *always* raise an `ExceptionGroup` (rather than, say, a single
-`ValueError`). Otherwise, it would be easy to accidentally write something like ``except
-ValueError:`` (not ``except*``), which works if a single exception is raised but fails to
-catch _anything_ in the case of multiple simultaneous exceptions (even if one of them is
-a ValueError). However, this is not how Trio worked in the past: as a concession to
-practicality when the ``except*`` syntax hadn't been dreamed up yet, the old
-``trio.MultiError`` was raised only when at least two exceptions occurred
-simultaneously. Adding a layer of `ExceptionGroup` around every nursery, while
-theoretically appealing, would probably break a lot of existing code in practice.
-
-Therefore, we've chosen to gate the newer, "stricter" behavior behind a parameter
-called ``strict_exception_groups``. This is accepted as a parameter to
-:func:`open_nursery`, to set the behavior for that nursery, and to :func:`trio.run`,
-to set the default behavior for any nursery in your program that doesn't override it.
-
-* With ``strict_exception_groups=True``, the exception(s) coming out of a nursery will
-  always be wrapped in an `ExceptionGroup`, so you'll know that if you're handling
-  single errors correctly, multiple simultaneous errors will work as well.
-
-* With ``strict_exception_groups=False``, a nursery in which only one task has failed
-  will raise that task's exception without an additional layer of `ExceptionGroup`
-  wrapping, so you'll get maximum compatibility with code that was written to
-  support older versions of Trio.
-
-The default is set to ``strict_exception_groups=True``, in line with the default behaviour
-of ``TaskGroup`` in asyncio and anyio.  We've also found that non-strict mode makes it
-too easy to neglect the possibility of several exceptions being raised concurrently,
-causing nasty latent bugs when errors occur under load.
+    For more on this pattern, see https://github.com/python-trio/trio/issues/2929
+    and the linked issue on trio-websocket.  We may want to provide a nursery mode
+    which handles this automatically; it's annoying but not too complicated and
+    seems like it might be a good feature to ship for such cases.
+
+**Third and most often**, the existence of a nursery in your code is not just an
+implementation detail, and callers *should* be prepared to handle multiple exceptions
+in the form of an `ExceptionGroup`, whether with ``except*`` or manual inspection
+or by just letting it propagate to *their* callers.  Because this is so common,
+it's nurseries' default behavior and you don't need to do anything.
```

Comment:
most often -> most common

But if we're talking about an API, I don't think it's desirable for this to be the most common.  I'm afraid it will become the most common though, because it takes a lot of work to do otherwise.  I'd guess that an `async` call or context manager having concurrency inside is usually an implementation detail, and should not be exposed.  Exposing ExceptionGroup from an API is a significant burden for users, and should be considered carefully.

Reply:
For many functions, internal concurrency is part of the documented semantics - for example it's not just an implementation detail that [`trio.serve_listeners()`](https://trio.readthedocs.io/en/stable/reference-io.html#trio.serve_listeners) has a nursery internally.  In such cases I think that responsibility for handling an `ExceptionGroup` _should_ lie with the caller.

I agree it's a fuzzy boundary though, and I'll remove the most-common claim.  I don't think the burden is particularly heavy once you're on Python 3.11+ and can use `except*`, though obviously this will vary by project.

Reply:
No doubt there are API's that need to expose concurrency.

A few burden points I can think of:
  * "once you're on Python 3.11+" - for any package trying to support the active Python versions, that's 2 years off.  In the meantime you need the cumbersome exceptiongroup.  And on the other hand, people will complain if you depend on exceptiongroup for 3.11+ even though it's harmless, complicating pinned dependencies for CI, etc...
  * we've agreed that pytest doesn't have a good story for matching the semantics of `except*`
  * that the user has to learn `except*` at all, even for an API that doesn't otherwise expose concurrency